static/js/containers/ProfilePage_test.js

```diff
@@ -70,6 +91,181 @@ describe("ProfilePage", function() {
+      return renderComponent(pageUrlStubs[2]).then(([, div]) => {
+        let toggle = div.querySelector('#profile-tab-professional-switch');
+
+        return listenForActions([SET_SHOW_WORK_DELETE_ALL_DIALOG], () => {
+          TestUtils.Simulate.change(toggle);
+          let dialog = openDialog();
+          let cancelButton = dialog.querySelector('.cancel-button');
+          TestUtils.Simulate.click(cancelButton);
+          let state = helper.store.getState();
+          let profile = state.profiles.jane.profile;
+          assert.equal(_.isEmpty(profile.work_history), false);
+        });
+      });
+    });
+
+    it('should confirm and let you delete when toggling the switch on work history', () => {
+      return renderComponent(pageUrlStubs[2]).then(([, div]) => {
+        let updateProfile = _.cloneDeep(USER_PROFILE_RESPONSE);
+        updateProfile.username = SETTINGS.username;
+        updateProfile.work_history = [];
+
+        patchUserProfileStub.throws("Invalid arguments");
+        patchUserProfileStub.withArgs(SETTINGS.username, updateProfile).returns(
+          Promise.resolve(updateProfile)
+        );
+
+        let toggle = div.querySelector('#profile-tab-professional-switch');
+        return listenForActions([
+          SET_SHOW_WORK_DELETE_ALL_DIALOG,
+          START_PROFILE_EDIT,
+          UPDATE_PROFILE_VALIDATION,
+          REQUEST_PATCH_USER_PROFILE,
+        ], () => {
+          TestUtils.Simulate.change(toggle);
+          let dialog = openDialog();
+          let deleteButton = dialog.querySelector('.delete-button');
+          TestUtils.Simulate.click(deleteButton);
+        });
+      });
+
+    });
+
+    it('shouldnt confirm when toggling the switch on work history if there are no entries', () => {
+      return renderComponent(pageUrlStubs[2]).then(([, div]) => {
+        let emptyWorkHistory = Object.assign({}, USER_PROFILE_RESPONSE, {
+          work_history: []
+        });
+        helper.store.dispatch(receiveGetUserProfileSuccess(SETTINGS.username, emptyWorkHistory));
+
+        let toggle = div.querySelector('#profile-tab-professional-switch');
+
+        return listenForActions([
+        ], () => {
+          TestUtils.Simulate.change(toggle);
+          assert.equal(openDialog(), undefined);
+          TestUtils.Simulate.change(toggle);
+          assert.equal(openDialog(), undefined);
+        });
+
+      });
+    });
+
+
+    let educationSwitchSelectors = EDUCATION_LEVELS.map(level => (
+      { value: level.value, label: level.label, selector: `#profile-tab-education-switch-${level.value}` }
+    ));
+
+    let educationEntries = (level, profile) => (
+      profile.education.filter(entry => entry.degree_name === level)
+    );
+
+    let fullEducation = () => {
+      let clone = _.cloneDeep(USER_PROFILE_RESPONSE);
+      [ASSOCIATE, MASTERS, DOCTORATE, MASTERS].forEach((level, index) => clone.education.push({
```

Comment:
Can you add two of each instead of just one, and add them such that they aren't right next to each other in the array? It would help testing that deletion works properly